applications/serial_lte_modem/doc/TCPIP_AT_commands.rst

```diff
@@ -0,0 +1,1541 @@
+------------
+
+The test command tests the existence of the command and provides information about the type of its subparameters.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPCLI: (op list),<url>,<port>,<sec_tag>
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xtcpcli=?
+   #XTCPCLI: (0, 1, 2),<url>,<port>,<sec_tag>
+   OK
+
+TCP send data #XTCPSEND
+=======================
+
+The ``#XTCPSEND`` command allows you to send the data over the connection.
+
+Set command
+-----------
+
+The set command allows you to send the data over the connection.
+When used from a TCP/TLS client, it sends the data to the remote TCP server
+When used from a TCP server, it sends data to the remote TCP client
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPSEND=<datatype>,<data>
+
+* The ``<datatype>`` parameter can accept one the following values:
+
+  * ``0`` - hexidecimal string (e.g. "DEADBEEF" for 0xDEADBEEF)
+  * ``1`` - plain text (default value)
+  * ``2`` - JSON
+  * ``3`` - HTML
+  * ``4`` - OMA TLV
+
+* The ``<data>`` parameter is a string and it contains the data being sent.
+  The maximum size for ``NET_IPV4_MTU`` is 576 bytes.
+  It should have no ``NULL`` character in the middle.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSEND: <size>
+
+* The ``<size>`` value is an integer and it represents the actual number of the bytes sent.
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xtcpsend=1,"Test TLS client"
+   #XTCPSEND: 15
+   OK
+
+Read command
+------------
+
+The read command is not supported.
+
+Test command
+------------
+
+The test command is not supported.
+
+TCP receive data #XTCPRECV
+==========================
+
+The ``#XTCPRECV`` command allows you to receive data over the connection.
+
+Set command
+-----------
+
+The set command allows you to receive data over the connection.
+It receives data buffered in the Serial LTE Modem.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPRECV[=<size>]
+
+* The ``<size>`` value is an integer and it represents the requested number of bytes.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   <data>
+   #XTCPRECV: <size>
+
+* The ``<size>`` value is an integer and it represents the actual number of the bytes received in the response.
+
+Read command
+------------
+
+The read command is not supported.
+
+Test command
+------------
+
+The test command is not supported.
+
+UDP server #XUDPSVR
+===================
+
+The ``#XUDPSVR`` command allows you to start and stop the UDP server.
+
+Set command
+-----------
+
+The set command allows you to start and stop the UDP server.
+
+Syntax
+~~~~~~
+
+::
+
+   #XUDPSVR=<op>[,<port>]
+
+* The ``<op>`` parameter can accept one the following values:
+
+  * ``0`` - Stop server
+  * ``1`` - Start server
+  * ``2`` - Start server with data mode support
+
+* The ``<port>`` parameter is an integer and it represents the UDP service port.
+  It is mandatory for starting the server.
+  The data mode is enabled when the TCP/TLS server is started.
+  Refer to ``#XSOCKET`` for the ``<sec-tag>`` and ``<peer_verify>`` definitions.
```

Comment:
Could you provide additional details on peer_verify? Also, should that be added to XSOCKET too? There is no mention of this in the list of XSOCKET parameters.

Reply:
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/bsdlib/include/nrf_socket.h#L156
Whether TLS/DTLS client need to verify the certificate from server or not.
Support in XSOCK TLS/DTLS client is not there yet. Could add if really needed

Reply:
@junqingzou What about in XUDPSVR? Is the peer verify parameter supported there?

Reply:
Peer_verify only applies to TLS/DTLS client/server.
There is no this feature supported for DTLS client yet (XUDPCLI).
And SLM does not support DTLS Server (modem limitation).